### PR TITLE
done

### PR DESCRIPTION
**Pull request by Richaa2**

### CODE REVIEW

File: src/main/java/practice/CandidateValidator.java [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+
+public class CandidateValidator implements Predicate<Candidate> {
+    @Override
+    public boolean test(Candidate candidate) {
+        return candidate.getAge() >= 35
+                && candidate.isAllowedToVote()
+                && candidate.getNationality().equals("Ukrainian")
+                && (Integer.parseInt(candidate.getPeriodsInUkr().split("-")[1])
+                - Integer.parseInt(candidate.getPeriodsInUkr().split("-")[0]))
+                >= 10;
```
Comment (aleksandr-bevza) [inline src/main/java/practice/CandidateValidator.java:14]: Move: 35, "Ukrainian", 1, 0, 10 to constants

File: src/main/java/practice/StreamPractice.java [hunk 2]
```diff
@@ -14,7 +19,14 @@ public class StreamPractice {
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(s -> s.split(","))
+                .flatMap(Arrays::stream)
```
Comment (aleksandr-bevza) [inline src/main/java/practice/StreamPractice.java:24]: You can do it in one line:
```suggestion
                .flatMap(s -> Stream.of(s.split(",")))
```

File: src/main/java/practice/StreamPractice.java [hunk 3]
```diff
@@ -59,7 +86,13 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        List<String> collect = peopleList.stream()
+                .filter(x -> x.getSex() == Person.Sex.WOMAN && x.getAge() >= femaleAge)
+                .map(Person::getCats)
+                .flatMap(Collection::stream)
```
Comment (aleksandr-bevza) [inline src/main/java/practice/StreamPractice.java:92]: You can do it in one line
```suggestion
                .flatMap(person -> person.getCats().stream())
```

File: src/main/java/practice/StreamPractice.java [hunk 4]
```diff
@@ -74,7 +107,16 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
...
      */
+
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return candidates.stream().filter(candidate -> candidate.getAge() >= 35
+                        && candidate.isAllowedToVote()
+                        && candidate.getNationality().equals("Ukrainian")
+                        && (Integer.parseInt(candidate.getPeriodsInUkr().split("-")[1])
+                        - Integer.parseInt(candidate.getPeriodsInUkr().split("-")[0]))
+                        >= 10)
```
Comment (aleksandr-bevza) [inline src/main/java/practice/StreamPractice.java:117]: You have created CandidateValidator predicate. You can use it in your filter
```suggestion
        return candidates.stream()
        .filter(new CandidateValidator())
```

**Review by aleksandr-bevza — CHANGES_REQUESTED** [review]

**Review by zachar-chorny — APPROVED** [review]

**Pull request closed without merging**